modules/UI/videolayout/VideoLayout.js

```diff
@@ -410,9 +410,7 @@ const VideoLayout = {
 
             // Enable the currently set dominant speaker.
             if (currentDominantSpeaker) {
-                if (smallVideo && smallVideo.hasVideo()) {
-                    this.updateLargeVideo(currentDominantSpeaker);
-                }
+                this.updateLargeVideo(currentDominantSpeaker);
             } else {
```

Comment:
In updateLargeVideo there are some smallVideo operations which are not protected, which were by this if is that ok? 
Not sure also whether if not smallVideo.hasVideo() then is smallVideo.videoStream defined.

Reply:
1. There is one issue with checking (smallVideo && smallVideo.hasVideo()). Small video refers to the video that was just unpinned, not necessarily the currentDominantSpeaker that will be switched to. I'm therefore unsure of the benefit of checking if the small video that was just clicked has video when updating large video.
2. VideoLayout#handleVideoThumbClicked calls VideoLayout.updateLargeVideo(currentDominantSpeaker). If the small video is on large video already, no large video update occurs. If the small video is not on large video, then LargeVideoManager#updateLargeVideo is called, which calls LargeVideoManager#scheduleLargeVideoUpdate. In that promise chain there is a variable isVideoMuted which checks for the stream existence and eventually sets the variable showAvatar to true.

I'm not seeing a problem but my code comprehension is not infallible. If there are some specific flows you'd like me to look at or test manually I'd love to do so.